src/ci/tested_versions_utils.py

```diff
@@ -1,38 +1,180 @@
 from __future__ import annotations
 
 import os
+import re
 from contextlib import contextmanager
 from dataclasses import dataclass
-from typing import Tuple
+from typing import List, Union
+
+# Major, minor, patch and (non semver standard, suffix)
+_semanticVersionPattern = re.compile(r'(!)?(\d+).(\d+).(\d+)([^\s]*)(?:\s*#\s*(.*))?')
+_splitVersionFromCommentPattern = re.compile(r'(!)?([^\s]*)(?:\s*#\s*(.*))?')
```

Comment:
Can you change it to capital letters with underscores separating words? https://peps.python.org/pep-0008/#constants